Documentation/project-docs/benchmarking.md

```diff
@@ -82,10 +82,65 @@ You should notice that given `.dll` file have been updated in the `CoreRun` fold
+
+After that, you should be able to find `System.Private.Corelib.dll` in a location similar to:
+
+        C:\Projects\coreclr\bin\Product\Windows_NT.x64.Release
+
+2. Build entire CoreFX in Release using your local private build of coreclr (See [Testing With Private CoreCLR Bits](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md#testing-with-private-coreclr-bits))
+
+        C:\Projects\corefx>build.cmd -release /p:CoreCLROverridePath=C:\Projects\coreclr\bin\Product\Windows_NT.x64.Release
+
+After that, you should be able to find `CoreRun.exe` in a location similar to:
+
+        C:\Projects\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\9.9.9\CoreRun.exe
+
+3. Create a new .NET Core console app using your favorite IDE
+4. Install BenchmarkDotNet (0.11.1+)
+5. Define the benchmarks and pass the arguments to BenchmarkSwitcher
+
+```cs
+class Program
+{
+   static void Main(string[] args) => BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);
+}
+```
+6. Run the benchmarks using `--coreRun` from the second step. Save the results in a dedicated folder.
+
+        dotnet run -c Release -f netcoreapp3.0 -- -f * --coreRun "C:\Projects\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\9.9.9\CoreRun.exe" --artifacts ".\before"
+
+7. Go to the corresponding CoreCLR source folder where the API you want to change exists (for example `coreclr\src\System.Private.CoreLib\shared\System`)
+8. Apply the optimization that you want to test
+9. Rebuild System.Private.Corelib with your change (optionally adding `-skipnative` if the change is isolated to managed code):
+
+        C:\Projects\coreclr>build.cmd -release -skiptests -skipnative
+
+10. For the next step you have one of two options depending:
+
+  - Rebuild given CoreFX part in Release:
+
+          C:\Projects\corefx>build.cmd -release /p:CoreCLROverridePath=C:\Projects\coreclr\bin\Product\Windows_NT.x64.Release
+
+  - OR manually copy over the relevant files from within the root of the coreclr output folder to where `CoreRun.exe` lives within corefx (excluding the subdirectories). This ends up being much faster than the first option and if the only thing that changed is   `System.Private.Corelib.dll`, just copy that over:
+
+          Copy from: C:\Projects\coreclr\bin\Product\Windows_NT.x64.Release
+          To: C:\Projects\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\9.9.9\
+
+11. Run the benchmarks using `--coreRun` from the first step. Save the results in a dedicated folder.
+
+        dotnet run -c Release -f netcoreapp3.0 -- -f * --coreRun "C:\Projects\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\9.9.9\CoreRun.exe" --artifacts ".\after"
+
+12. Compare the results and repeat steps `8 - 11` until you are happy about the results.
```

Comment:
There's a tool for comparing the results, right?

Reply:
cc @adamsitnik, do we have such a tool?